### PR TITLE
Feature/directives&pipes homework

### PR DESCRIPTION
**Pull request by lessarea**

### CODE REVIEW

File: src/app/courses/course/duration.pipe.spec.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { DurationPipe } from './duration.pipe';
+
+describe('DurationPipe', () => {
+  const pipe = new DurationPipe();
```
Comment (andrewlistopadov) [inline src/app/courses/course/duration.pipe.spec.ts:4]: What does your pipe do? It represents time formatting passed number to a string.
What if I pass falsy value or not a number?

File: src/app/courses/course/status-shadow.directive.spec.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { Component } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { StatusShadowDirective } from './status-shadow.directive';
+
+@Component({
+  template: `
+  <h2 [agmpStatusShadow]="'04/25/2020'">Upcoming. With green shadow</h2>
```
Comment (andrewlistopadov) [inline src/app/courses/course/status-shadow.directive.spec.ts:8]: The same as above, there are no negative scenarios.

File: src/app/courses/courses-search/courses-search.component.ts [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
 
 @Component({
   selector: 'courses-search',
   templateUrl: './courses-search.component.html',
   styleUrls: ['./courses-search.component.scss'],
 })
 export class CoursesSearchComponent implements OnInit {
+  @Output() search = new EventEmitter<string>();
   public searchValue: string;
```
Comment (andrewlistopadov) [inline src/app/courses/courses-search/courses-search.component.ts:10]: You don't need an extra field here. You can get the search value from the event.

File: src/app/courses/courses-search/courses-search.component.html [hunk 4]
```diff
@@ -2,7 +2,7 @@
   <mat-form-field appearance="outline" class="search__input">
     <mat-icon matPrefix class="icon big search__icon">search</mat-icon>
     <mat-label>Text to search</mat-label>
-    <input matInput [(ngModel)]="searchValue">
+    <input matInput [(ngModel)]="searchValue" (keypress)="handleKeyPress($event)">
```
Comment (andrewlistopadov) [inline src/app/courses/courses-search/courses-search.component.html:5]: You don't need ngModel in your particular case here.
You are able to get the value from the event.

File: src/app/courses/courses-search/courses-search.component.ts (hunk 3 above)
Comment (lessarea) [reply src/app/courses/courses-search/courses-search.component.ts]: Replaced `ngModel` with form to cover both (click and enter) submit cases

File: src/app/courses/course/status-shadow.directive.spec.ts (hunk 2 above)
Comment (lessarea) [reply src/app/courses/course/status-shadow.directive.spec.ts:8]: done

File: src/app/courses/course/duration.pipe.spec.ts (hunk 1 above)
Comment (lessarea) [reply src/app/courses/course/duration.pipe.spec.ts:4]: done

File: src/app/courses/course/course.component.ts [hunk 5]
```diff
@@ -8,6 +8,7 @@ import { ICourse } from '../shared/course.model';
   styleUrls: ['./course.component.scss'],
 })
 export class CourseComponent implements OnInit {
+  @Input() ngClass: string | string[] | Set<string> | { [klass: string]: any; };
```
Comment (andrewlistopadov) [inline src/app/courses/course/course.component.ts:11]: You don't need it. ngClass is a build it angular directive. The only thing you need to use it on the template level is CommonModule importing.

**Pull request merged**